src/services/ui_manager.cpp

```diff
@@ -156,4 +157,60 @@ namespace otto::services {
     last_frame = now;
   }
 
+  void UIManager::register_screen_key(core::input::Key key, ScreenEnum se)
+  {
+    Controller::current().register_key_handler(
+      key,
+      [this, se] {
+        reset_timer();
+        display(se);
+      },
+      [this] {
+        if (timer_done()) {
+          display(pop_back());
+        } else {
+          push_back(state.current_screen);
+        }
+      });
+  }
+  void UIManager::register_screen_key(core::input::Key key, ScreenEnum s, ScreenEnum sm, core::input::Key modifier)
+  {
+    Controller::current().register_key_handler(
+      key,
+      [this, s, sm, modifier] {
+        reset_timer();
+        if (Controller::current().is_pressed(modifier)) {
+          display(sm);
+        } else {
+          display(s);
+        }
+      },
+      [this] {
+        if (timer_done()) {
+          display(pop_back());
+        } else {
+          push_back(state.current_screen);
+        }
+      });
+  }
```

Comment:
so, this is not what i meant, but it might be an ok solution. I wanted to do basically this, but as a general solution in `Controller` where `KeyReleaseHandler` takes a bool saying whether it was a long press. However, that probably requires storing a timer per key or something like that, so this is fine for now.

Reply:
I know, I had to adapt it a bit. Because there are buttons that are not screen buttons, I felt that was the best place to put it. No reason for `KeyReleaseHandler` to look at this for every button released.

Reply:
it would probably be a nice feature to have in other places as well, but yes, lets just keep it here for now. Remember, this is the logic thread, we are a lot less pressed on performance